src/check_missing_alleles.py

```diff
@@ -0,0 +1,439 @@
+
+    imgt_dict = {}
+
+    # Get the mapping between allele name and IMGT accession
+    with open("./Allelelist.txt") as fh:
+        # Skip IMGT header
+        for _ in range(6):
+            next(fh)
+        rows = csv.DictReader(fh, delimiter=",")
+        for row in rows:
+            allele = row["Allele"]
+            allele = (":").join(allele.split(":")[:2])
+            if allele not in imgt_dict and "N" not in allele and "Q" not in allele:
+                imgt_dict[allele] = row["AlleleID"]
+
+    # Find differences between IMGT and MRO alleles
+    imgt_alleles = set(imgt_dict.keys())
+    missing_alleles = imgt_alleles.difference(mro_alleles)
+
+    missing_allele_rows = set()
+    for allele in missing_alleles:
+        try:
+            accession = imgt_dict[allele]
+            source = "IMGT/HLA"
+            label = f"HLA-{allele} chain"
+            resource_name = allele
+            seq = seqs[imgt_dict[allele]]
+            missing_allele_rows.add((label, resource_name, source, accession, seq))
+        except Exception:
+            continue
+
+    with open("../ontology/chain-sequence.tsv", "a+") as outfile:
+        for tup in missing_allele_rows:
+            outfile.write(("\t").join(tup) + "\n")
+
+    return missing_alleles
+
+
+def update_chain(missing_alleles):
+    """Update the chain.tsv table to add new chains from IMGT
+
+    Returns:
+        A list of missing genes ['DRB', 'Y']
+    """
+
+    missing_genes = set()
+    for allele in missing_alleles:
+        gene = allele.split("*")[0]
+        missing_genes.add(gene)
+
+    mro_chains = set()
+    mro_genes = set()
+    with open("../ontology/chain.tsv") as fh:
+        rows = csv.DictReader(fh, delimiter="\t")
+        for row in rows:
+            label = row["Label"]
+            if "HLA" in label:
+                if "*" not in label:
+                    gene = label.split(" ")[0][4:]
+                    mro_genes.add(gene)
+                else:
+                    allele = label.split(" ")[0][4:]
+                    mro_chains.add(allele)
+
+    new_genes = missing_genes.difference(mro_genes)
+    new_gene_tups = set()
+    for gene in new_genes:
+        label = f"HLA-{gene} chain"
+        synonyms = ""
+        class_type = "equivalent"
+        parent = "protein"
+        gene = f"HLA-{gene} locus"
+        new_gene_tups.add((label, synonyms, class_type, parent, gene))
+
+    new_chains = missing_alleles.difference(mro_chains)
+    new_chain_tups = set()
+    for allele in new_chains:
+        label = f"HLA-{allele} chain"
+        synonyms = ""
+        class_type = "subclass"
+        # This removes the specific allele # (generic chain)
+        parent = f"HLA-{allele.split('*')[0]} chain"
+        gene = ""
+        new_chain_tups.add((label, synonyms, class_type, parent, gene))
+
+    with open("../ontology/chain.tsv", "a+") as fh:
+        for tup in new_gene_tups:
+            fh.write(("\t").join(tup) + "\n")
```

Comment:
`csv.writer(fh, delimiter="\t", lineterminator="\n")` might be a better choice here